### PR TITLE
Use citus_finish_citus_upgrade() in the tests

### PR DESCRIPTION
**Pull request by onderkalaci**
We already have tests relying on citus_finalize_upgrade_to_citus11().
Now, adjust those to rely on citus_finish_citus_upgrade() and
always call citus_finish_citus_upgrade().

On top of #5995, opened for ease of review

### CODE REVIEW

File: src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+NOTICE:  Preparing all the existing partitioned table indexes
+NOTICE:  Preparing to sync the metadata to all nodes
+ROLLBACK;
+-- do the actual job
+CALL citus_finish_citus_upgrade();
+NOTICE:  Preparing all the existing partitioned table indexes
+NOTICE:  Preparing to sync the metadata to all nodes
+-- idempotent, should be called multiple times
+CALL citus_finish_citus_upgrade();
+NOTICE:  already at the latest distributed schema version (11.1-1)
```
Comment (marcocitus) [inline src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out:14]: wil this become something we have to change in the future?

File: src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- Citus upgrades are finished by calling a procedure
+-- this is a transactional procedure, so rollback should be fine
+BEGIN;
+	CALL citus_finish_citus_upgrade();
+NOTICE:  Preparing all the existing partitioned table indexes
+NOTICE:  Preparing to sync the metadata to all nodes
+ROLLBACK;
+-- do the actual job
+CALL citus_finish_citus_upgrade();
```
Comment (marcocitus) [inline src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out:9]: Wondering whether we should check the pg_dist_node_metadata upgrade_version

File: src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out (hunk 1 above)
Comment (onderkalaci) [reply src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out:14]: That is true, it'll change. I can increase the log level to warning here

File: src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out (hunk 2 above)
Comment (onderkalaci) [reply src/test/regress/expected/upgrade_citus_finish_citus_upgrade.out:9]: makes sense

**Review by marcocitus — APPROVED** [review]

**Pull request merged**